### ISSUE TITLE
Root note is sometimes unsteady with cv modulation

### ISSUE THREAD

**Issue by Normalised** [TomWhitwell/RM_Chorder#12]
Comment on muff's : 

https://www.muffwiggler.com/forum/viewtopic.php?p=2643653#2643653

My chord organ changes its root note from time to time. This happens either when I have lfo patched into root note or cv sequence. Usually, the root note holds for some time than it changes returns back etc... It doesn't happen when I don't have anything plugged into root note. On the other hand, the root note doesn't changes when I'm modulating chord. 

I've uploaded audio from it for better understanding: 
Chord organ issue 

https://hearthis.at/tasun/chord-organ-issue/ 

In this case I'm modulating root note with voltage block, you can hear it in 5th second.

**Comment by Normalised** [comment]
I'll expose the input smoothing boundary size to the config, see if that can be tweaked to resolve this case.

**Comment by meeeerz** [comment]
I think I'm having a similar issue. When I send v/Oct from a sequencer or keyboard an extra note is quickly triggered before returning to the correct note. It doesn't occur consistently.

I can make a video later if that would help.

**Comment by Thonk-Steve** [comment]
Hey, yeah a video would be good. Have you ever tried it on a
different eurorack case or PSU?

**Comment by meeeerz** [comment]
I have not. I'll move it to where I've got a music thing radio music that's working perfectly when I try to recreate the issue.

A 4ms row 40 is usually powering my little trunk case racks. Will update later.

**Comment by Thonk-Steve** [comment]
aha, well one thing you could try also is temporarily programming your
Radio Music to a Chord Organ to see if you can replicate it there.

You could also see if for example putting the radio music front PCB on the
Chord organ back pcb/teensy solves the problem. I would just label the back
PCBs and teensy so you can remember which is which before you start doing
that :)

**Comment by meeeerz** [comment]
https://photos.app.goo.gl/8yx1aj2owD5rcGWx6

https://photos.app.goo.gl/7jgmsc9gUQnRBamG8

I was also able to recreate the problem with other v/Oct devices. On to switching around pcbs!

**Comment by meeeerz** [comment]
Okay loaded up the other teensy with firmware 0.6.
Same exact issue. This time the video is sequenced using a beatstep pro.
I checked my boards and I can't find any problems in the construction. Radio music works great on the one in this vid.

Gonna try older chord organ firmware now?

https://photos.app.goo.gl/ufx6YNe3d36U4ZzHA

**Comment by rackd0** [comment]
Hit the same/similar issue on a new CO module a few days ago (not sure if there's already a published fix - here's what worked for me).
When I compiled Chord-Organ 0.6 with some debugging ouput, the value read from the CV pin for a fixed voltage seemed to vary quite widely, although I didn't try too hard to pin down the cause.  Instead, I tweaked Analoginput.h to set the booleans "hysteresis" and "average" to true.  I bumped up the value of "borderThreshold" from 16 to a much higher value.  I tried 100 initially, after weighing up the variance I was seeing, the ADC resolution, and the range being quantised.  Notes are now rock-steady (sincere thanks Normalised!).

**Comment by Alexcoleman268** [comment]
Hey Everyone, i am having the same exact issue, ive tried putting it in another case but that was all i could do @rackd0 i cant seem to understand what you are saying to do either. Any help would be greatly appreciated.

**Comment by rackd0** [comment]
Hey Alex – are you set up to be able to load a new version of firmware onto your Chord Organ?

**Comment by Alexcoleman268** [comment]
@rackd0 yeah i am! what should i do, i have already loaded the most recent firmware. I also have a friend with the same exact problem.

**Comment by rackd0** [comment]
@Alexcoleman268 Provided you can reload a known-good firmware should things go awry, see if this fixes the issue for you...

[Chord_Organ_0.6_cv.zip](https://github.com/TomWhitwell/Chord-Organ/files/7189546/Chord_Organ_0.6_cv.zip)

**Comment by Alexcoleman268** [comment]
@rackd0 Hey That did it!! Thaks alot!!!

**Comment by rackd0** [comment]
@Alexcoleman268 Glad to hear that - you're most welcome!  Huge thanks to @Normalised who suggested this approach (input smoothing).  Possibly there is a later firmware where this can be tweaked via the config, as he describes (comment 8 Oct 2017).

**Comment by Buenyam1** [comment]
Hey guys, I had the same issue sending stepped cv from voltera. Thank you @rackd0 for sharing the other firmware
